cmake/modules/FindMPI.cmake

```
@@ -124,9 +124,48 @@ IF(NOT DEFINED MPI_VERSION OR MPI_VERSION STREQUAL ".")
   SET(MPI_VERSION_MINOR "0")
 ENDIF()
 
+#
+# Make sure that we do not run into underlinking on Debian/Ubuntu systems with
+# lld / ld.gold and missing libopen-pal.so on the link line:
+#
+
+CHECK_COMPILER_SETUP(
+  "${DEAL_II_CXX_FLAGS_SAVED} ${DEAL_II_CXX_FLAGS}"
+  "${DEAL_II_LINKER_FLAGS_SAVED} ${DEAL_II_LINKER_FLAGS}"
+  MPI_UNDERLINKAGE_OK
+  ${MPI_CXX_LIBRARIES} ${MPI_Fortran_LIBRARIES} ${MPI_C_LIBRARIES}
+  )
+
+IF(NOT MPI_UNDERLINKAGE_OK)
+  MESSAGE(STATUS "Try to avoid underlinkage by expliclitly adding libopen-pal to link line")
```

This sounds like an instruction to the user. Maybe this?
```suggestion
  MESSAGE(STATUS "Trying to avoid underlinkage by expliclitly adding libopen-pal to link line")
```